### PR TITLE
Persist navigator data across technologies

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 91188362

## Summary

If there is an item in the navigator, that links to another technology/documentation, going back should still re-apply the old filters.

## Dependencies

https://github.com/apple/swift-docc-render/pull/234

## Testing

Steps:
1. Assert there are no regressions when working with normal doccarchives.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/Navigator/NavigatorCard.vue [hunk 1]
```diff
@@ -835,7 +859,16 @@ export default {
     setActiveUID(uid) {
       this.activeUID = uid;
       this.resetScroll = false;
-      sessionStorage.set(STORAGE_KEYS.activeUID, uid);
+    },
+    /**
+     * Handles the `navigate` event from NavigatorCardItem, guarding from selecting an item,
+     * that points to another technology.
+     */
+    handleNavigationChange(uid) {
```
Comment (marinaaisa) [inline src/components/Navigator/NavigatorCard.vue:867]: If a user doesn't navigate through the technology (they go straight away to a new different technology instead by just filtering) we never set the activeUID and therefore we cannot persist the filtering data :(

We need a solution also for those cases.

**Review by marinaaisa — CHANGES_REQUESTED** [review]
Left couple of comments. One is about a use case that it's still not persisting the data but maybe that's why the PR is still on draft. :)

File: src/components/Navigator/NavigatorCard.vue (hunk 1 above)
Comment (dobromir-hristov) [reply src/components/Navigator/NavigatorCard.vue:866]: It was called, but I had some mistakes in my logic. Should be fixed now.

**Review by dobromir-hristov — COMMENTED** [review]
Try again, should be fixed now.

**Review by marinaaisa — APPROVED** [review]
Testing works great and code too. Let me know again to give another look when you finish the PR and open it (if it's not finished yet)

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**